src/common/components/select-repository-list/index.js

```diff
@@ -32,7 +33,9 @@ export class SelectRepositoryListComponent extends Component {
   componentWillReceiveProps(nextProps) {
     const repositoriesStatus = nextProps.repositoriesStatus;
     const ids = Object.keys(repositoriesStatus);
+    const owner = nextProps.user.login;
     if (ids.length && ids.every((id) => repositoriesStatus[id].success)) {
+      this.props.dispatch(fetchUserSnapsIfNeeded(owner));
```

Comment:
This feels a bit like a workaround rather then solution, as it's quite clear that the problem is in the dashboard (we trigger redirect before we properly receive the data).

I had a look at the component flow there and indeed it's tricky. The `fetchUserSnaps` action is dispatched (so in theory it sets `isFetching` to true and `success` to false in the store), but the component still calls `fetchData` with its current state (success with zero repositories).

So it seem that the problematic line is this one:
https://github.com/canonical-ols/build.snapcraft.io/blob/master/src/common/components/repositories-home/index.js#L42

Removing it seems to fix the bug. By removing it we never trigger `fetchData` directly from `componentDidMount`, only from `componentWillRecieveProps`. In `componentDidMount` we always fetch user snaps, so eventually store will be updated and component will receive new props so it can decide if redirect should be made or builds should be fetched.

Unless I missed something and it will break in some other case...